src/MessageEvent.php

```diff
@@ -29,6 +29,8 @@ public static function parse($data)
                 $message->lastEventId .= $value;
             } elseif ($name === 'event') {
                 $message->type = $value;
+            } elseif ($name === 'retry' && preg_match('/^\d+$/', $value)) {
+                $message->retry = $value < PHP_INT_MAX ? (int)$value : (float)$value;
```

Comment:
How about `(string) (int) $value !== $value && $value >= 0`?

Reply:
Also, if my calculation isn't off then even a 32 bit system can hold a maximum of ~25 days in this field. I don't think there are many use cases that need a float here.

Related tests can probably be simplified by using a structure similar to the existing tests in `MessageEventTest` :+1: